### PR TITLE
Add video streaming whiteboarding problem

### PR DESCRIPTION
**Pull request by shamamaz**
Added in a rough draft of a whiteboarding problem, for a video streaming site.

Would appreciate some help on specific API endpoints or more specific questions we can add to this.

### CODE REVIEW

File: whiteboard.md [hunk 1]
```diff
@@ -20,3 +20,14 @@ An API service will be used to consume and create content from this database. De
 - Create a new article for an issue, e.g. `POST /articles` or another possibility `POST /issues/X/Y/articles`
 - Update an article, e.g. `PUT/PATCH /articles/:id`. Converation point: PUT vs PATCH. Candidate should be aware of the common update patterns
 - Delete an article, e.g. `DELETE /articles/:id`
+
+### Online Streaming Service
```
Comment (nicolechung) [inline whiteboard.md]: Can we add numbers to the problems, i.e. #1 and #2?

File: whiteboard.md [hunk 2]
```diff
@@ -20,3 +20,14 @@ An API service will be used to consume and create content from this database. De
...
+#### Overview
+We require a video streaming site, which will display tv shows and movies to users. The primary feature on this site will be its "Recommended for you" section, which will be based on the viewing history and ratings that the user provides for the videos that they watch. Metadata including categories, title tags, length, studio, and director information, will be stored on each video. It will be used to primarily separate the site into various categories, which the user may be recommended as a whole as well.
```
Comment (nicolechung) [inline whiteboard.md]: `We require a video streaming site`

Suggestion: could be something simpler, like:

`You've been asked to build a video streaming site`

The overview has a lot of requirements, for easier reading I suggest breaking this single paragraph down into several sentences, with bullets for the listed items (i.e. recommended for you, metadata):

```
The primary feature on this site will be its "Recommended for you" section, which will be based on:
- user's viewing history 
- ratings that the user provides for the videos that they watch. 

Metadata, will be stored on each video:
- categories
- title tags
- length
- studio
- director information.

The metadata will be used to primarily separate the site into various categories, which the user may be recommended as a whole as well. 
```

File: whiteboard.md [hunk 3]
```diff
@@ -20,3 +20,14 @@ An API service will be used to consume and create content from this database. De
...
 - Update an article, e.g. `PUT/PATCH /articles/:id`. Converation point: PUT vs PATCH. Candidate should be aware of the common update patterns
 - Delete an article, e.g. `DELETE /articles/:id`
+
+### Online Streaming Service
+#### Overview
+We require a video streaming site, which will display tv shows and movies to users. The primary feature on this site will be its "Recommended for you" section, which will be based on the viewing history and ratings that the user provides for the videos that they watch. Metadata including categories, title tags, length, studio, and director information, will be stored on each video. It will be used to primarily separate the site into various categories, which the user may be recommended as a whole as well. 
+
+##### Database Design
+- Design a DB schema that captures the domain above.
+-- Mentor should guide conversation, including discussions around optimized storage for large files, and how metadata and how to optimize internet streaming speed, as well as retrieve and categorize the information from the metadata.
```
Comment (nicolechung) [inline whiteboard.md]: How much about video streaming and transcoding will the student be expected to know to whiteboard this problem? 🤔

**Comment by nicolechung** [comment]
I added more feedback than's Affaf's primarily because its the weekend and I have time now.

File: whiteboard.md (hunk 3 above)
Comment (nicolechung) [reply whiteboard.md]: e.g. Will the questions be like:
- How would you stream this on a slow 3G connection vs high-speed DSL?

Comment (shamamaz) [reply whiteboard.md]: I envision the questions could go either way, based on both student goals and skill level, in addition to mentor expertise. A student applying for a more junior position will be required to have some working knowledge, or at the very least an the ability to critically think and articulate some thoughts on most general questions.

**Comment by shamamaz** [comment]
Will this whiteboarding.md doc be shared with students, or will the mentor simply read/use this as a guideline to interview? The mentor notes are visible inline.

**Comment by nicolechung** [comment]
@shamamaz good question! I believe we will just keep these notes for the mentors but maybe we can make a slide for the overview part to display, thoughts?

**Pull request merged**